.github/workflows/build-and-publish.yaml

```diff
@@ -29,11 +39,56 @@ jobs:
         if: startsWith('env.BRANCH_PREFIX', env.GITHUB_REF)
         run: |
           BRANCH_NAME=${GITHUB_REF#refs/heads/}
-          echo "::set-output name=version::${BRANCH_NAME/${{ env.BRANCH_PREFIX }}/}"
-
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-
+          echo "version=${BRANCH_NAME/${{ env.BRANCH_PREFIX }}/}" >> $GITHUB_OUTPUT
+      
+      - name: Install IBMCLI
+        run: curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+          
+      - name: IBM Cloud Login
+        run: ibmcloud login -q --apikey ${{ secrets.IBMCLOUD_API_KEY }} -r eu-gb 
+          
+      - name: Install VPC CLI plug-in
+        run: ibmcloud plugin install vpc-infrastructure
+          
+      - name: Creation of ZVSI
+        id: ZVSI
+        run: |
+          #creation of zvsi
+          ibmcloud is instance-create $ZVSI_INSTANCE_NAME $ZVSI_VPC $ZVSI_ZONE_NAME $ZVSI_PROFILE_NAME $ZVSI_SUBNET --image $ZVSI_IMAGE --keys $ZVSI_KEY --resource-group-id $ZVSI_RG_ID
+          #Reserving a floating ip to the ZVSI
+          ibmcloud is floating-ip-reserve $ZVSI_FP_NAME --zone $ZVSI_ZONE_NAME --resource-group-id $ZVSI_RG_ID --in $ZVSI_INSTANCE_NAME
+          #Bouding the Floating ip to the ZVSI
+          ibmcloud is floating-ip-update $ZVSI_FP_NAME --nic primary --in $ZVSI_INSTANCE_NAME
+          sleep 20
+            
+          #Saving the Floating IP to login ZVSI
+          ZVSI_HOST=$(ibmcloud is floating-ip $ZVSI_FP_NAME | awk '/Address/{print $2}')
+          echo $ZVSI_HOST
+          echo "IP=${ZVSI_HOST}" >> $GITHUB_OUTPUT
+          
+      - name: Setup Docker on ZVSI
+        uses: appleboy/ssh-action@master
+        with:
+            host: ${{ steps.ZVSI.outputs.IP }}
+            username: ${{ secrets.SSH_USER }}
+            key: ${{ secrets.ZVSI_PR_KEY }}
+            script: |
+              #update
+              arch
+              apt-get update
+              
+              #Setup Docker
+              apt-get install ca-certificates curl gnupg
+              install -m 0755 -d /etc/apt/keyrings
+              curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+              chmod a+r /etc/apt/keyrings/docker.gpg
+              echo \
+              "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+              "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+              tee /etc/apt/sources.list.d/docker.list > /dev/null
+              apt-get update
+              apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin -y
```

Comment:
Creating a custom image/snapshot where this is already set up will help reduce the build time.

Reply:
yeah, that will be a good option that reduces the build time. will update the custom image which is pre-installed with docker.

Thanks for the tip ✌️